### PR TITLE
[Bugfix] Trim spaces in forum thread titles

### LINKED ISSUE #3519: No Validation for blank input
ezgif com-video-to-gif(9)

### PR DESCRIPTION
**Pull request by drumilpatel2000**
fixes #3519 
ezgif com-video-to-gif(10)

### CODE REVIEW

**Comment by MasterOdin** [comment]
While I could agree with trimming the title, I'm not sure there's never a use case an instructor might not use spaces at the beginning of a post for formatting reasons.

**Comment by drumilpatel2000** [comment]
@MasterOdin Please look at this one 
ezgif com-video-to-gif(11)

**Comment by MasterOdin** [comment]
Discussing with @bmcutler and @andrewaikens87, it seems like instructors do rely on that they can just input a single space in a thread body and then use that purely as the body. Both instructors and students have utilized this feature for good so we'd prefer not to take this away, so let's make this feature only affect titles.

File: site/public/js/server.js [hunk 1]
```diff
@@ -2618,6 +2618,11 @@ function toggleSidebar() {
 }
 
 $(document).ready(function() {
+    $("textarea, input[type='text']").change(function () {
+        if($.trim($(this).val()).length == 0){
+            $(this).val($.trim($(this).val()));
+        }
+    })
```
Comment (MasterOdin) [inline site/public/js/server.js]: add a colon here

**Comment by drumilpatel2000** [comment]
@MasterOdin  Is this okay?

**Comment by MasterOdin** [comment]
@drumilpatel2000 This is fine and will be merged asap.

File: site/app/templates/forum/ThreadPostForm.twig [hunk 2]
```diff
@@ -127,4 +127,11 @@
...
 </div>
 <br>
-</div>
\ No newline at end of file
+</div>
+<script>
+    $("input[name='title']").change(function () {
+        if($.trim($(this).val()).length == 0){
+            $(this).val($.trim($(this).val()));
+        }
```
Comment (MasterOdin) [inline site/app/templates/forum/ThreadPostForm.twig]: Let's get rid of the `if` statement and just use the `$(this).val($.trim($(this).val()));` bit to always trim it. We don't ever want the spaces on the start/end of the string.

**Pull request merged**